tests/scenario_tests/test_function.py

```diff
@@ -168,9 +170,34 @@ def test_function_handler_timeout(self, monkeypatch):
         assert response.status == 404
         assert_auth_test_count(self, 1)
         assert (
-            sleep_mock.call_count == 5
+            sleep_mock.call_count == timeout
         ), f"Expected handler to time out after calling time.sleep 5 times, but it was called {sleep_mock.call_count} times"
 
+    def test_warning_when_timeout_improperly_set(self, caplog):
+        app = App(
+            client=self.web_client,
+            signing_secret=self.signing_secret,
+        )
+        app.function("reverse")(just_no_ack)
+        assert "WARNING" not in caplog.text
+
+        timeout_argument_name = "acknowledgement_timeout"
+        kwargs = {timeout_argument_name: 5}
+
+        callback_id = "reverse1"
+        app.function(callback_id, **kwargs)(just_no_ack)
+        assert (
+            f'WARNING On @app.function("{callback_id}"), as `auto_acknowledge` is `True`, `{timeout_argument_name}={kwargs[timeout_argument_name]}` you gave will be unused'
+            in caplog.text
+        )
```

Comment:
quibble: Hardcoding the expected outputs might make this output more immediate at a glance! I'm not so familiar with `kwargs` I must admit 😔

Reply:
I used `kwargs` in the test in order to assert that the parameter name shown in the warning is the one defined by the function 🤔 I think we should leave it for now but we don't need to make this a common pattern